### PR TITLE
added ability to modify nova flavor metadata in os_nova_flavor module

### PR DESCRIPTION
**Pull request by kuboj**
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
os_nova_flavor

##### ANSIBLE VERSION
```
ansible 2.2.0.0
```

##### SUMMARY
This PR adds ability to specify metadata (aka extra_specs) for nova flavors using shade API.

### CODE REVIEW

File: lib/ansible/modules/cloud/openstack/os_nova_flavor.py [hunk 1]
```diff
@@ -88,6 +88,11 @@
           assigned if a value is not specified.
      required: false
      default: "auto"
+   extra_specs:
+     description:
+        - Metadata dictionary
+     required: false
+     default: None
```
Comment (gundalow) [inline lib/ansible/modules/cloud/openstack/os_nova_flavor.py:93]: Please add `version_added: "2.3"` to this section

File: lib/ansible/modules/cloud/openstack/os_nova_flavor.py [hunk 2]
```diff
@@ -107,6 +112,18 @@
     cloud: mycloud
     state: absent
     name: tiny
+
+# Create flavor with metadata
```
Comment (gundalow) [inline lib/ansible/modules/cloud/openstack/os_nova_flavor.py]: Please use `- name: Create flavor with metadata` (and remove `-` from in front of `os_nova_flavor:`

**Review by gundalow — CHANGES_REQUESTED** [review]
Have a look at docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html for details on how to document modules

**Comment by kuboj** [comment]
Documentation fixed according to comments by @gundalow

**Comment by kuboj** [comment]
ready_for_review

File: lib/ansible/modules/cloud/openstack/os_nova_flavor.py (hunk 1 above)
Comment (rcarrillocruz) [inline lib/ansible/modules/cloud/openstack/os_nova_flavor.py]: Along with the doc change, please force the version requirement in the code itself too.
We use StrictVersion for that, have a look at https://github.com/ansible/ansible/blob/829c0b8f6284ec37e36f1454d7d3640415239cb2/lib/ansible/modules/cloud/openstack/os_network.py#L186 .

**Comment by rcarrillocruz** [comment]
shipit

**Review by gundalow — APPROVED** [review]

**Pull request merged**